### DISCUSSION TITLE
自定义组件

### DISCUSSION THREAD

**Discussion by fongfai** [alibaba/formily#1897] [Q&A 2.x] [answered]
designable/antd/src/schemas/CSSStyle.ts  请问下这个文件里 比如 DisplayStyleSetter   BackgroundStyleSetter  等这些x-component 是如何渲染的  因为formily自带的组件库里没有这些组件， 也没有查到自定义开发这些组件的代码

**Comment by janryWang** [comment, marked as answer]
在designable源码里
